packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -1237,19 +1238,24 @@ export default class TradeStore extends BaseStore {
             updateChartType,
             updateGranularity,
         } = this.root_store.contract_trade || {};
-        if (isAccumulatorContract(obj_new_values.contract_type) || isDigitTradeType(obj_new_values.contract_type)) {
+        const has_line_chart =
+            isAccumulatorContract(obj_new_values.contract_type) ||
+            isDigitTradeType(obj_new_values.contract_type) ||
+            isUpDownContract(obj_new_values.contract_type);
+        if ((chart_type !== 'line' || granularity !== 0) && has_line_chart) {
             savePreviousChartMode(chart_type, granularity);
             updateGranularity(0);
             updateChartType('line');
         } else if (
-            (obj_new_values.contract_type || obj_new_values.symbol) &&
```

Comment:
Will it still work upon `symbol` (asset) change without `|| obj_new_values.symbol`? )

Reply:
yeah, production logic is broken without `|| obj_new_values.symbol` check:
You may notice in your test link that when you are on `AUD/JPY + Multipliers + 2 min` chart and you switch to `Volatility + Accumulators`( the chart will auto-switch to `1 tick`), then if you go back to `AUD/JPY` and `Multipliers` will get auto-selected, the chart will be still `1 tick` instead of `2 min`.

Reply:
Seems like the logic is completely broken in general on production, once Accumulators is opened, it will never switch back from ticks to anything on other trade types. No idea how to fix it, it's one of the most horrible code I've seen in dtrader. Regarding `obj_new_values.symbol`, I really have no idea why it was there, this part of logic is not supposed to be executed when symbol changes. Symbol has nothing to do with chart type and granularity. My logic works fine if the first trade type you open is anything other than 1 tick ones, f.e. if you open vanillas or turbos first. Making the default trade type is accumulators basically breaks the whole logic as chart type and granularity immediately saved and set to localStorage because of appearing in URL

Reply:
I will refactor this whole thing later